### PR TITLE
base configure

### PR DESCRIPTION
**Pull request by mm25zamanian**

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "version": "0.0.0",
+  "npmClient": "yarn",
+  "useWorkspaces": true,
+  "package": [
+    "api",
```
Comment (asmanp2012) [inline lerna.json:6]: Rename api to demo and make demo folder.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  "license": "MIT",
+  "type": "module",
+  "private": true,
+  "devDependencies": {
+    "lerna": "^4.0.0",
+    "typescript": "^4.5.4"
+  },
+  "workspaces": [
+    "package/*",
+    "api"
```
Comment (asmanp2012) [inline package.json:16]: Rename api to demo and make demo folder

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    /* Experimental Options */
+    "experimentalDecorators": true,              /* Enables experimental support for ES7 decorators. */
+    "emitDecoratorMetadata": true,               /* Enables experimental support for emitting type metadata for decorators. */
+
+    /* Advanced Options */
+    "skipLibCheck": true,                           /* Skip type checking of declaration files. */
+    "forceConsistentCasingInFileNames": true,        /* Disallow inconsistently-cased references to the same file. */
+  },
+  "include": [ "src/**/*.ts" ],
```
Comment (asmanp2012) [inline tsconfig.json:77]: Add demo like above

File: lerna.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      "hoist": true,
+      "npmClientArgs": [
+        "--no-package-lock"
+      ]
+    },
+    "run": {
+      "stream": true
+    }
+  }
+}
```
Comment (asmanp2012) [inline lerna.json:31]: Add a breakline at the end of all files.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  ]
+}
```
Comment (asmanp2012) [inline package.json:18]: Add a breakline at the end of all files.

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  "exclude": [],
+  "references": [
+    // Add Packages
+  ]
+}
```
Comment (asmanp2012) [inline tsconfig.json:82]: Add a breakline at the end of all files.

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+node_modules/
```
Comment (asmanp2012) [inline .gitignore:1]: Complete with sample

```
# Specifies intentionally untracked files to ignore when using Git
# http://git-scm.com/docs/gitignore

# OSX
.DS_STORE
.DS_Store
._*

# Windows
Thumbs.db
Desktop.ini

# Logs
*.log*

# Build
dist
node_modules
.tsbuildinfo
tsconfig.tsbuildinfo
package-lock.json

*.js
*.map
*.d.ts
```

File: package.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "one-web-components",
```
Comment (asmanp2012) [inline package.json:2]: change that to @one/core

File: package.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  "version": "0.1.0",
```
Comment (asmanp2012) [inline package.json:3]: Change that to 0.0.0

File: package.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+  "description": "One Web Components",
+  "repository": "https://github.com/AliMD/One",
+  "author": "Ali Mihandoost <alimihandoost@gmail.com>",
+  "license": "MIT",
```
Comment (asmanp2012) [inline package.json:7]: Add contributors.

**Review by asmanp2012 — CHANGES_REQUESTED** [review]
1. Add a breakline at the end of all files  And.
2. Rename api to demo and make demo folder.

**Comment by mm25zamanian** [comment]
Everything was done
Ok ??

File: .gitignore [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
```
Comment (asmanp2012) [inline .gitignore:1]: Remove this line.

Comment (mm25zamanian) [reply .gitignore]: Done.
I forgot to delete this 😂

File: lerna.json [hunk 12]
```diff
@@ -0,0 +1,24 @@
+{
+  "version": "0.0.0",
+  "npmClient": "yarn",
+  "useWorkspaces": true,
+  "package": ["demo", "package/*"],
+  "command": {
+    "version": {
+      "conventionalCommits": true,
+      "changelogPreset": "lit",
```
Comment (asmanp2012) [inline lerna.json:9]: change that to "angular"

File: package.json [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+  "name": "@one/core",
+  "version": "0.0.0",
+  "description": "One Web Components",
+  "repository": "https://github.com/AliMD/One",
+  "author": "Ali Mihandoost <alimihandoost@gmail.com>",
+  "contributors": [
+    "MohammadMahdi Zamanian <mm25zamanian@gmail.com>"
+  ],
+  "license": "MIT",
+  "type": "module",
```
Comment (asmanp2012) [inline package.json:11]: Remove this line

File: package.json [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+  "private": true,
+  "devDependencies": {
+    "lerna": "^4.0.0",
+    "typescript": "^4.5.4"
+  },
+  "workspaces": [
+    "package/*",
+    "demo"
```
Comment (asmanp2012) [inline package.json:19]: demo/*

File: tsconfig.json [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+    "tsBuildInfoFile": ".tsbuildinfo",                     /* Specify file to store incremental compilation information */
+    "removeComments": true,                         /* Do not emit comments to output. */
+    // "noEmit": true,                              /* Do not emit outputs. */
+    "importHelpers": true,                          /* Import emit helpers from 'tslib'. */
+    // "downlevelIteration": true,                  /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
+    // "isolatedModules": true,                     /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
+
+    /* Strict Type-Checking Options */
+    "strict": true,                                 /* Enable all strict type-checking options. */
+    "noImplicitAny": true,                       /* Raise error on expressions and declarations with an implied 'any' type. */
```
Comment (asmanp2012) [inline tsconfig.json:35]: why `"noImplicitAny": true,` uncomment ??

**Review by asmanp2012 — CHANGES_REQUESTED** [review]
Remove extra config

File: tsconfig.json (hunk 15 above)
Comment (mm25zamanian) [reply tsconfig.json:35]: We want to use typescript rules in a strict framework

Comment (mm25zamanian) [reply tsconfig.json:35]: So our variables should not be in the state of anything

Comment (asmanp2012) [reply tsconfig.json:35]: A few codes like below, maybe not work in this config. it's better handled by Eslint that we fix it in another pull request.

```
(x as any).inssert()
```
comment this line.

**Comment by njfamirm** [comment]
gooooooooooodrat 😆

**Pull request merged**